src/overlaybd/tar/erofs/liberofs.cpp

```diff
@@ -0,0 +1,661 @@
+#include "liberofs.h"
+#include "erofs/tar.h"
+#include "erofs/io.h"
+#include "erofs/cache.h"
+#include "erofs/block_list.h"
+#include "erofs/inode.h"
+#include "../../lsmt/file.h"
+#include "../../lsmt/index.h"
+#include <photon/common/alog.h>
+
+#define SECTOR_SIZE 512ULL
+#define SECTOR_BITS 9
+#define round_down_blk(addr) ((addr) & (~(SECTOR_SIZE - 1)))
+#define round_up_blk(addr) (round_down_blk((addr) + SECTOR_SIZE - 1))
+#define min(a, b) (a) < (b) ? (a) : (b)
+#define MAP_FILE_NAME "upper.map"
+
+#define EROFS_UNIMPLEMENTED 1
+
+struct erofs_sector {
```

Comment:
I think we could just use a std::map<addr, struct liberofs_inmem_sector> and stl::set for dirty bitmap?

struct liberofs_inmem_sector {
     char data[SECTOR_SIZE];
};

Reply:
Done.